source/_localCaptioner/__init__.py

```diff
@@ -0,0 +1,194 @@
+
+	Args:
+		captioner: The captioner instance to use for generation.
+		imageData: The image data to caption.
+	"""
+	try:
+		description = captioner.generate_caption(image=imageData)
+		ui.message(description)
+		api.copyToClip(text=description, notify=False)
+	except FileNotFoundError as e:
+		ui.message(str(e))
+	except Exception as e:
+		log.error(e)
+
+
+class LocalCaptioner:
+	"""module for local image caption functionality.
+
+	This module provides image captioning using local ONNX models.
+	It can capture screen regions and generate descriptive captions.
+	"""
+
+	def __init__(self) -> None:
+		"""Initialize the global plugin."""
+		# super().__init__()
+		self.isModelLoaded = False
+		self.captioner: ImageCaptioner | None = None
+
+		loadModelWhenInit = config.conf["captionLocal"]["loadModelWhenInit"]
+		# Load model when initializing  (may cause high memory usage)
+		if loadModelWhenInit:
+			threading.Thread(target=self._loadModel, daemon=True).start()
+
+	def terminate(self):
+		self.captioner = None
+
+	def runCaption(self, gesture) -> None:
+		# def script_runCaption(self) -> None:
+		"""Script to run image captioning on the current navigator object.
+
+		Args:
+			gesture: The input gesture that triggered this script.
+		"""
+		imageData = shootImage()
+
+		if not self.isModelLoaded:
+			# Translators: Message when loading the model
+			ui.message(_("loading model..."))
+			self._loadModel()
+
+		imageThread = threading.Thread(target=caption, args=(self.captioner, imageData))
+		# Translators: Message when starting image recognition
+		ui.message(_("starting recognize"))
+		imageThread.start()
+
+	def _loadModel(self) -> None:
+		"""Load the ONNX model for image captioning.
+
+		Raises:
+			Exception: If the model cannot be loaded.
+		"""
+		try:
+			localModelDirPath = config.conf["captionLocal"]["localModelPath"]
+			encoderPath = f"{localModelDirPath}/onnx/encoder_model_quantized.onnx"
+			decoderPath = f"{localModelDirPath}/onnx/decoder_model_merged_quantized.onnx"
+			configPath = f"{localModelDirPath}/config.json"
+
+			self.captioner = ImageCaptioner(
+				encoder_path=encoderPath,
+				decoder_path=decoderPath,
+				config_path=configPath,
+			)
+			self.isModelLoaded = True
+		except Exception as e:
+			self.isModelLoaded = False
+			ui.message(str(e))
+			raise
+
+	def releaseModel(self, gesture) -> None:
+		"""Script to release the loaded model from memory.
+
+		Args:
+			gesture: The input gesture that triggered this script.
+		"""
+		self.doReleaseModel()
+
+	def doReleaseModel(self) -> None:
+		# Translators: Message when releasing the model
+		ui.message(_("releasing model..."))
+		try:
+			if hasattr(self, "captioner") and self.captioner:
+				del self.captioner
+				self.captioner = None
+				# Translators: Message when model is successfully released
+				ui.message(_("model released and memory freed"))
+				self.isModelLoaded = False
+		except Exception as e:
+			ui.message(str(e))
```

Comment:
Please make this a user friendly, translated string. Don't message python exceptions to end users

Reply:
@tianzeshi-study - you marked this is as resolved but this is not. Please make this a translated, user friendly string e.g. "failed to release model". Do not use `ui.message` on a python exception.